TaskManager/Modules/Home/View/HomeViewController.swift

```diff
@@ -102,14 +131,17 @@ extension HomeViewController: UITableViewDelegate {
             return UISwipeActionsConfiguration()
         }
         let deleteAction = UIContextualAction(style: .destructive, title: "Delete") { (action, sourceView, completionHandler) in
-            
-            var snapshot = self.dataSource.snapshot()
-            snapshot.deleteItems([item])
-            self.dataSource.apply(snapshot, animatingDifferences: true)
+            self.presenter.deleteTask(item)
 
             completionHandler(true)
         }
-        let editAction = UIContextualAction(style: .normal, title: "Edit") { _,_,_  in
+        let editAction = UIContextualAction(style: .normal, title: "Edit") { _,_,completionHandler  in
+            
+            let presenter = self.presenter.dataStorage
```

Comment:
because homeViewController will never get deallocated you will never have memory leaks here but usually its better to use weak self instead of strong to prevent memory leaks when the object can get deallocated